presto-main/src/main/java/com/facebook/presto/util/maps/IdentityLinkedHashMap.java

```diff
@@ -142,4 +133,116 @@ public int size()
             }
         };
     }
+
+    private class KeySet
```

Comment:
In #7354 i created `IndetityLinkedHashMap columnReferences` and wrote `columnReferences.keySet().contains(..)` -- equality comparison was used.

(Of course, I should write `columnReferences.containsKey(..)` but that's not the point)